value_test.go

```diff
@@ -348,6 +349,27 @@ func TestMap(t *testing.T) {
 	assert.InDelta(t, float64(33), value.(map[string]any)["audience1"], floatEpsilon)
 }
 
+func TestDecimal(t *testing.T) {
+	_, conn := SetupTestDatabase(t)
+	res, error := conn.Query("UNWIND [1] AS A UNWIND [5.7, 8.3, 8.7, 13.7] AS B WITH cast(CAST(A AS DECIMAL) * CAST(B AS DECIMAL) AS DECIMAL(18, 1)) AS PROD RETURN COLLECT(PROD) AS RES")
+	assert.Nil(t, error)
+	assert.True(t, res.HasNext())
+	next, _ := res.Next()
+	value, _ := next.GetValue(0)
+	size := len(value.([]interface{}))
+	assert.Equal(t, 4, size)
+
+	expected := decimal.NewFromFloat(5.7)
+	assert.Equal(t, expected, value.([]interface{})[0])
+	expected = decimal.NewFromFloat(8.3)
+	assert.Equal(t, expected, value.([]interface{})[1])
+	expected = decimal.NewFromFloat(8.7)
+	assert.Equal(t, expected, value.([]interface{})[2])
+	expected = decimal.NewFromFloat(13.7)
+	assert.Equal(t, expected, value.([]interface{})[3])
+	expected = decimal.NewFromFloat(8.7 * 13.7)
```

Comment:
There are multiple calls to `value.([]interface{})`. This is not readable and redundent. You should assign it to a var and use it for the following assertions.